.github/workflows/ci.yml

```diff
@@ -51,15 +51,27 @@ jobs:
           - os: macos-12
             python-version: [ 3, 11 ]
             pip-version: 20
+            tox-env-python: python
           - os: ubuntu-22.04
             python-version: [ 3, 11 ]
             pip-version: 20
+            tox-env-python: python
           - os: ubuntu-22.04
             python-version: [ 3, 11 ]
             pip-version: 22_3_1
+            tox-env-python: python
           - os: ubuntu-22.04
             python-version: [ 3, 11 ]
             pip-version: 23_1_2
+            tox-env-python: python
+          - os: macos-12
+            python-version: [ 3, 12, "0-beta.3" ]
+            pip-version: 23_2
+            tox-env-python: python3.11
```

Comment:
It turned out tox doesn't work (or I couldn;t get it to work) with 3.12; thus this complication. More generally, trying to support a new Python version, especially when it yanks core stdlib like distutils, is hard if you're early. Eventually tox will support 3.12 and this complication will be removed.